micro_speech/rp2/audio_provider.cc

```diff
@@ -151,37 +156,43 @@ TfLiteStatus GetAudioSamples(tflite::ErrorReporter* error_reporter,
   // often enough and the buffer is large enough that this call will be made
   // before that happens.
 
-  const int16_t kAdcSampleDC = 2048;
-  const int16_t kAdcSampleGain = 20;
+  const int16_t kAdcSampleDC = 0; // this is done in capture samples now
+  const int16_t kAdcSampleGain = 1; // set to 1 incase it wants including
 
   // Determine the index, in the history of all samples, of the first
   // sample we want
   //printf("getting audio samples\n");
   const int start_offset = start_ms * (kAudioSampleFrequency / 1000);
   // Determine how many samples we want in total
   const int duration_sample_count = duration_ms * (kAudioSampleFrequency / 1000);
-  for (int i = 0; i < duration_sample_count; ++i) {
-    // For each sample, transform its index in the history of all samples into
-    // its index in g_audio_capture_buffer
-    const int capture_index = (start_offset + i) % kAudioCaptureBufferSize;
-    const int32_t capture_value = g_audio_capture_buffer[capture_index];
-    int32_t output_value = capture_value - kAdcSampleDC;
-    //
-    output_value *= kAdcSampleGain;
-    //printf("%d, \n", output_value);
-    if (output_value < std::numeric_limits<int16_t>::min()) {
-      output_value = std::numeric_limits<int16_t>::min();
-    }
-    if (output_value > std::numeric_limits<int16_t>::max()) {
-      output_value = std::numeric_limits<int16_t>::max();
-    }
-    // Write the sample to the output buffer
-    g_audio_output_buffer[i] = output_value;
-    //printf("%d\n", output_value);
-    //g_audio_output_buffer[i] = capture_value;
+
+  //loop not required anymore
+  //for (int i = 0; i < duration_sample_count; ++i) {
+  //  // For each sample, transform its index in the history of all samples into
+  //  // its index in g_audio_capture_buffer
+  //  const int capture_index = (start_offset + i) % kAudioCaptureBufferSize;
+  //  const int32_t capture_value = g_audio_capture_buffer[capture_index];
+  //  int32_t output_value = capture_value - kAdcSampleDC;
+  //  //
+  //  output_value *= kAdcSampleGain;
+  //  //printf("%d, \n", output_value);
+  //  if (output_value < std::numeric_limits<int16_t>::min()) {
+  //    output_value = std::numeric_limits<int16_t>::min();
+  //  }
+  //  if (output_value > std::numeric_limits<int16_t>::max()) {
+  //    output_value = std::numeric_limits<int16_t>::max();
+  //  }
+  //  // Write the sample to the output buffer
+  //  g_audio_output_buffer[i] = output_value;
+  //  //printf("%d\n", output_value);
+  //  //g_audio_output_buffer[i] = capture_value;
+  //}
+  for (int i =0; i < NSAMP; i ++) {
```

Comment:
Could consider using `memcpy` here.

Reply:
@sandeepmistry yes I think this part needs thinking through. It use to add values from a larger array into a smaller array, but now it adds the whole array. So the loop could be taken out entirely